# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: fbf55a44d7fb
# Full Hash: fbf55a44d7fb72908eeb5d041b70ce2c503a2532
# Author: Anny Gakhokidze <agakhokidze@mozilla.com>
# Date: 2020-04-15 04:09:17
# Regressor Bug: 1594529
# File Overlap Count: 1
# Description:
#   Bug 1594529 - Create LoadInfo for subdocuments directly in parent process with DocumentChannel. r=mattwoodrow,nika
#   
#   Currently, with Fission enabled we are not able to create a proper LoadInfo
#   object when doing a subdocument load because we do not have access to a loading
#   context if the load is happening inside of an OOP frame. To solve this problem,
# ==============================================================================

diff -r 4ba9a230586a -r fbf55a44d7fb dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Mon Apr 13 22:41:37 2020 +0000
+++ b/dom/ipc/PWindowGlobal.ipdl	Mon Apr 13 23:00:43 2020 +0000
@@ -12,6 +12,7 @@
 
 include DOMTypes;
 include ClientIPCTypes;
+include NeckoChannelParams;
 
 using JSWindowActorMessageKind from "mozilla/dom/JSWindowActor.h";
 using mozilla::gfx::IntRect from "mozilla/gfx/Rect.h";
@@ -83,6 +84,27 @@
   /// Update the URI of the document in this WindowGlobal.
   async UpdateDocumentURI(nsIURI aUri);
 
+  // We expose frameAncestors to web-extensions and they extract URIs from the
+  // principals collected. In order to be compatible with that API, we need to
+  // update the document's principal. This is only allowed if the principals are
+  // `equals` to each other.
+  async UpdateDocumentPrincipal(nsIPrincipal aPrincipal);
+
+  // Update document's `documentHasLoaded` bit in this WindowGlobal.
+  async UpdateDocumentHasLoaded(bool aDocumentHasLoaded);
+
+  // Update document's 'documentHasUserInteracted' bit in this WindowGlobal.
+  async UpdateDocumentHasUserInteracted(bool aDocumentHasUserInteracted);
+
+  // Update document's sandbox flags in this WindowGlobal.
+  async UpdateSandboxFlags(uint32_t aSandboxFlags);
+
+  // Update document csp's fields in this WindowGlobal.
+  async UpdateDocumentCspSettings(bool aBlockAllMixedContent, bool aUpgradeInsecureRequests);
+
+  // Update document's cookie settings in this WindowGlobal.
+  async UpdateCookieJarSettings(CookieJarSettingsArgs cookieJarSettings);
+
   // Update the title of the document in this WindowGlobal.
   async UpdateDocumentTitle(nsString aTitle);
 